src/main/java/org/ibp/api/rest/germplasm/GermplasmResource.java

```diff
@@ -119,4 +153,33 @@ public long getFilteredCount() {
 		return new ResponseEntity<>(this.germplasmService.searchAttributes(query), HttpStatus.OK);
 	}
 
+	@RequestMapping(value = "/crops/{cropName}/germplam-sets/templates/xls", method = RequestMethod.GET)
+	public ResponseEntity<FileSystemResource> getImportGermplasmExcelTemplate(@PathVariable final String cropName,
+		@RequestParam(required = false) final String programUUID) {
+
+		final VariableFilter variableFilter = new VariableFilter();
+		variableFilter.addPropertyId(TermId.INVENTORY_AMOUNT_PROPERTY.getId());
+		final List<VariableDetails> units = this.variableService.getVariablesByFilter(variableFilter);
+
+		final List<LocationDto> storageLocation =
+			this.locationService.getLocations(cropName, programUUID, GermplasmResource.STORAGE_LOCATION_TYPE, null, null, false);
+
+		final List<LocationDto> locations =
+			this.locationService.getLocations(cropName, programUUID, GermplasmResource.LOCATION_TYPE, null, null, false);
+
+		final List<org.generationcp.middleware.api.brapi.v1.attribute.AttributeDTO> attributeDTOS = this.germplasmService.getGermplasmAttributes();
+
+		final List<Method> breedingMethods = this.germplasmService.getAllBreedingMethods();
+
+		final List<GermplasmName> germplasmNames = this.germplasmService.getGermplasmNames();
+
+		final File file =
+			this.germplasmTemplateExportService.export(breedingMethods, germplasmNames, attributeDTOS, locations, storageLocation, units);
```

Comment:
Is possible to get all these parameters inside `germplasmTemplateExportService.writeCodesSheet`?